.github/workflows/release.yml

```diff
@@ -0,0 +1,39 @@
+name: Release
+
+on:
+  workflow_run:
+    workflows: [ Test ]
+    types: [ completed ]
+    branches: [ master ]
+
+jobs:
+  deploy:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    environment:
+      name: release
+      url: https://pypi.org/p/mozdownload
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - name: Install dependencies
+        run: python -m pip install -U build
+
+      - name: Build a binary wheel and a source tarball
+        run: python -m build
+
+      - name: Upload release to PyPI
+        uses: pypa/gh-action-pypi-publish@release/v1
+  on-failure:
+    runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'failure' }}
+    steps:
```

Comment:
A quick follow-up question (otherwise all looks fantastic) to those lines, are they really needed? When a failure happens we will still see which specific job has failed so I don't think this will deliver more helpful data. I would just remove it. Also please keep an empty line at the end of the file.

Reply:
Those lines are not needed, but I just want to make the pipeline verbose, so I fixed the 
pipeline but I forgot to add a condition: `github.event.workflow_run.event == 'push'`. I think in case pull_request is not needed to upload the release.

Reply:
So the addition of `push` was probably not correct given that we always run the `release` workflow now. Instead we should limit this workflow to PRs that are about releases only. It's very unlikely that branches except for release use only version numbers so we could check `github.ref_name` for like starting with a number?

Reply:
Hello, I'll fix this

Reply:
Great. Given that this PR is closed lets follow-up on the related issue for now.